### PR TITLE
Allow non-V implementations of vector units, with Zve/Zvl extensions

### PR DESCRIPTION
**Pull request by jerryz123**
**Related issue**: 

**Type of change**: bug report | feature request | other enhancement

**Impact**: no functional change | API addition (no impact on existing code) | API modification

**Development Phase**: proposal |  implementation

**Release Notes**

### CODE REVIEW

File: src/main/scala/tile/BaseTile.scala [hunk 1]
```diff
@@ -104,8 +104,17 @@ trait HasNonDiplomaticTileParameters {
     val f = if (tileParams.core.fpu.nonEmpty) "f" else ""
     val d = if (tileParams.core.fpu.nonEmpty && tileParams.core.fpu.get.fLen > 32) "d" else ""
     val c = if (tileParams.core.useCompressed) "c" else ""
-    val v = if (tileParams.core.useVector) "v" else ""
+    val v = if (tileParams.core.useVector && tileParams.core.vLen >= 128 && tileParams.core.eLen == 64 && tileParams.core.vfLen == 64) "v" else ""
     val h = if (usingHypervisor) "h" else ""
+    val zvl = Option.when(tileParams.core.useVector) { Seq(s"zvl${tileParams.core.vLen}b") }
```
Comment (sequencer) [inline src/main/scala/tile/BaseTile.scala:109]: `vLen > 1024` is not defined in spec, although it also says:
>  Longer vector length extensions should follow the same pattern.

I still think the spec should explicitly define it.

File: src/main/scala/tile/BaseTile.scala [hunk 2]
```diff
@@ -104,8 +104,17 @@ trait HasNonDiplomaticTileParameters {
     val f = if (tileParams.core.fpu.nonEmpty) "f" else ""
     val d = if (tileParams.core.fpu.nonEmpty && tileParams.core.fpu.get.fLen > 32) "d" else ""
     val c = if (tileParams.core.useCompressed) "c" else ""
-    val v = if (tileParams.core.useVector) "v" else ""
+    val v = if (tileParams.core.useVector && tileParams.core.vLen >= 128 && tileParams.core.eLen == 64 && tileParams.core.vfLen == 64) "v" else ""
```
Comment (sequencer) [inline src/main/scala/tile/BaseTile.scala:107]: > The single-letter V extension is intended for use in application processor profiles.

Does spec also indicate privileged spec support, e.g. `usingUser`?

**Review by sequencer — APPROVED** [review]
Another my complain is [riscv-opcodes](https://github.com/riscv/riscv-opcodes) should use a fine-grained way to declare instructions(so we can parse it.)

File: src/main/scala/tile/BaseTile.scala (hunk 1 above)
Comment (jerryz123) [reply src/main/scala/tile/BaseTile.scala:109]: Take it up with the spec

File: src/main/scala/tile/BaseTile.scala (hunk 2 above)
Comment (jerryz123) [reply src/main/scala/tile/BaseTile.scala:107]: No.

**Pull request merged**